lib/components/src/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,207 @@
+  /** Set to true to add rounded borders from theme to Accordion */
+  rounded?: boolean;
+  /**
+   * Set to true to add theme borders between AccordionItem's
+   * Note: can be overridden on AccordionItem
+   */
+  lined?: boolean;
+  /**
+   * Set to true to indent AccordionBody content left to match the AccordionHeader text
+   * Note: can be overridden on AccordionItem
+   */
+  indentBody?: boolean;
+  /**
+   * Set to true to apply narrow paddings and smaller text in all AccordionItem's
+   * Note: can be overridden on AccordionItem
+   */
+  narrow?: boolean;
+  /**
+   * Set to true to prevent toggle between open and close for all AccordionItem's
+   * Note: can be overridden on AccordionItem
+   */
+  preventToggle?: boolean;
+  /** Courtesy callback prop for when a new accordion opens */
+  onOpen?: (change: StateChange) => void;
+  /** Courtesy callback prop for when an accordion closes */
+  onClose?: (change: StateChange) => void;
+} & React.HTMLAttributes<HTMLUListElement>;
+
+/**
+ *
+ * Controlling wrapper component that will control the open/close functionality to
+ * child ```<AccordionItem />```'s and style the other support components
+ * ```<AccordionHeader />``` and ```<AccordionBody />```
+ *
+ * Although each component can work independently and be controlled from the outside
+ * via an **open** prop, it is not recommended to use ```<AccordionItem />```,
+ *  ```<AccordionHeader />``` and ```<AccordionBody />``` without this controller
+ *
+ * If no ```<AccordionBody />``` is inside an ```<AccordionItem />``` the open/close
+ * functionality will be removed by default, as well as the icon and pointer cursor.
+ */
+export const Accordion: FC<AccordionProps> = ({
+  allowMultipleOpen,
+  children,
+  open,
+  rounded = false,
+  bordered = false,
+  narrow, // To be carried to AccordionItem children via context
+  lined, // To be carried to AccordionItem children via context
+  indentBody, // To be carried to AccordionItem children via context
+  preventToggle, // To be carried to AccordionItem children via context
+  onOpen,
+  onClose,
+  ...rest
+}) => {
+  const [openState, setOpenState] = useState<OpenMap>({});
+  const openStateMap = useRef<OpenMap>({});
+  const accordionMap = useRef<AccordionMap>({});
+  const accordionMapIndex = useRef(0);
+
+  const addToMap: AddToMapFn = useCallback(
+    (id) => {
+      const index = accordionMapIndex.current + 1;
+      accordionMap.current = { ...accordionMap.current, [id]: { id, index } };
+      accordionMapIndex.current = index;
+
+      if (typeof open === 'number') {
+        openStateMap.current = { ...openStateMap.current, [id]: open === index };
+      } else if (Array.isArray(open)) {
+        const isExpanded = open.includes(index);
+        openStateMap.current = { ...openStateMap.current, [id]: isExpanded };
+      } else {
+        openStateMap.current = { ...openStateMap.current, [id]: false };
+      }
+    },
+    [accordionMap, openStateMap, open]
```

Comment:
You're creating a new function every time the `open` prop changes

Reply:
Thanks man - will fix